src/main/java/com/fleet/status/controller/FleetStatusController.java

```diff
@@ -44,46 +46,49 @@ public String AircraftStatus() {
+            return new ResponseEntity<>(eventService.getHomepageAircraft(), HttpStatus.OK);
         } catch(Exception e) {
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
     @GetMapping("/getOutOfServiceAircraft")
     @ResponseBody
-    public ResponseEntity<List<Aircraft>> getOutOfServiceAircraft() {
+    public ResponseEntity<List<Event>> getOutOfServiceAircraft() {
         try {
-            return new ResponseEntity<>(aircraftService.getOutOfServiceAircraft(), HttpStatus.OK);
+            return new ResponseEntity<>(eventService.getOutOfServiceAircraft(), HttpStatus.OK);
         } catch(Exception e) {
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
-    @PostMapping(value="/addAircraftEvent", consumes = "application/json", produces = "application/json")
+    @PostMapping(value="/saveEvent", consumes = "application/json", produces = "application/json")
     @ResponseBody
-    public ResponseEntity<String> submitEvent(@RequestBody Aircraft aircraft) {
+    public ResponseEntity<String> submitEvent(@RequestBody Event event) {
         try {
-            aircraftService.save(aircraft);
-            log.info("Tail number {} saved.", aircraft.getTailNumber());
-            return new ResponseEntity<>("Aircraft saved.", HttpStatus.CREATED);
+            eventService.save(event);
+            return new ResponseEntity<>("New event saved.", HttpStatus.CREATED);
+        } catch (DataIntegrityViolationException e) {
+            return new ResponseEntity<>("Tail number already exists.", HttpStatus.CONFLICT);
         } catch (Exception e) {
-            log.error("Failed to save aircraft with tail number {}", aircraft.getTailNumber(), e);
-            return new ResponseEntity<>("Failed to save aircraft.", HttpStatus.INTERNAL_SERVER_ERROR);
+            log.error("Failed to save event for tail number {}", event.getAircraft().getTailNumber(), e);
+            return new ResponseEntity<>("Failed to save event.", HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
     @PostMapping(value = "/calcDowntime", consumes = "application/json", produces = "application/json")
     @ResponseBody
-    public String calcDowntime(@RequestBody int aircraftId) {
-        Aircraft aircraft = aircraftService.fetchById(aircraftId);
+    public String calcDowntime(@RequestBody int eventId) {
+        Event event = eventService.findById(eventId);
 
-        if (aircraft.getEndTime() == null || aircraft.getStartTime() == null) {
+        if (event.getEndTime() == null || event.getStartTime() == null) {
             return "Down time is not available";
         }
-        long hours = aircraftService.calculateDownTime(aircraft.getStartTime(), aircraft.getEndTime());
+
+        //long hours = eventService.calculateDownTime(event.getStartTime(), event.getEndTime());
+        long hours = 0;
```

Comment:
You might be able to just call the new getDownTIme() method you added. That might fix this endpoint since downtime on the AircraftStatus page is showing as 0 hours

Reply:
Also, if you go that route, you can remove the calculateDownTime method from IEventService & EventService. Won't need those anymore

Reply:
What do you think about removing down time from the Aircraft Status page all together?  It's actually pretty pointless since when you show an aircraft back in service it disappears off the Aircraft Status page.  You would never see the downtime anyway.  We could replace that with an edit button so user can go back and edit an event if they messed up a remark or the update time got pushed back.

Reply:
Yeah, removing that altogether does make a bit more sense. We can leave this as-is and have someone else implement that in a future story